src/App.js

```diff
@@ -21,15 +28,29 @@ export default function App() {
   return (
     <Switch>
       <Route exact path="/">
+        <MainPage onHandleChange={changeCategory} />
+      </Route>
+      <Route path="/suggestions">
+        <Suggestions filteredMovies={filteredMovies} />
+      </Route>
+      <Route path="/addmovie">
         <Toaster />
         <MovieFormPage onAddMovie={addMovie} />
       </Route>
       <Route path="/movies">
-        <MovieListPage movies={movies} />
+        <MovieListPage
+          filteredMovies={filteredMovies}
+          onHandleChange={changeCategory}
+        />
       </Route>
     </Switch>
   )
 
+  function changeCategory(e) {
```

Comment:
I think it's good to have these kind of functions, which update the state, in one place. Since we have our `return` statement before these functions it is still well structured.